matrix_sdk/src/client.rs

```diff
@@ -308,49 +306,44 @@ impl Client {
     /// * `homeserver_url` - The homeserver that the client should connect to.
     ///
     /// * `config` - Configuration for the client.
-    pub fn new_with_config<U: TryInto<Url>>(
+    pub fn new_with_client<U: TryInto<Url>>(
         homeserver_url: U,
         config: ClientConfig,
+        http_client: Arc<dyn HttpClient>,
```

Comment:
Why isn't this part of the `ClientConfig`?

Reply:
Right, that would make more sense with how the rest of it works (using the `ClientConfig` to pass in stuff for the `Client`). I'll rebase everything and update this PR to use the `ClientConfig`.